src/dex-ui/store/poolsSlice/poolsSlice.ts

```diff
@@ -27,6 +39,78 @@ const initialPoolsStore: PoolsState = {
   },
 };
 
+const fetchEachToken = (evmAddress: string) => {
```

Comment:
Please refactor this function using async/await and breakdown the function into a more modular and readable format.

Reply:
Done